src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState } from 'react';
+import cn from 'classnames';
+
+import { Todo } from '../../types/Todo';
+
+import { deleteTodo } from '../../api/todos';
+
+interface TodoItemProps {
+  todo: Todo;
+  filterTodoList: (todoId: number) => void;
+}
+
+export const TodoItem: React.FC<TodoItemProps> = ({
+  todo,
+  filterTodoList,
+}) => {
+  const [loading, setLoading] = useState(false);
+
+  const handleDeleteTodo = (todoId: number) => {
+    setLoading(true);
+    deleteTodo(todoId)
+      .then(() => {
+        setLoading(false);
+        filterTodoList(todoId);
+      });
```

Comment:
There's no error handling for the promise returned by deleteTodo function. If the promise fails, the loading state will not change and the user interface will not respond appropriately.